core/arch/x86/encode.c

```diff
@@ -2107,6 +2178,75 @@ encode_vex_final_prefix_byte(byte cur_byte, decode_info_t *di, const instr_info_
     return cur_byte;
 }
 
+static byte *
+encode_evex_prefixes(byte *field_ptr, decode_info_t *di, const instr_info_t *info,
+                     bool *output_initial_opcode)
+{
+    byte val = 0;
+    CLIENT_ASSERT(output_initial_opcode != NULL, "required param");
+    *output_initial_opcode = true;
+    *field_ptr = 0x62;
+    di->evex_encoded = true;
+    field_ptr++;
+    /* second evex byte */
+    val = /* these are negated */
+        (TEST(PREFIX_REX_R, di->prefixes) ? 0x00 : 0x80) |
+        (TEST(PREFIX_REX_X, di->prefixes) ? 0x00 : 0x40) |
+        (TEST(PREFIX_REX_B, di->prefixes) ? 0x00 : 0x20) |
+        (TEST(PREFIX_EVEX_RR, di->prefixes) ? 0x00 : 0x10);
+    if (TEST(OPCODE_THREEBYTES, info->opcode)) {
+        byte op3 = (byte)((info->opcode & 0x00ff0000) >> 16);
+        if (op3 == 0x38)
+            val |= 0x02;
+        else if (op3 == 0x3a)
+            val |= 0x03;
+        else
+            CLIENT_ASSERT(false, "unknown 3-byte opcode");
+    } else {
+        byte op3 = (byte)((info->opcode & 0x00ff0000) >> 16);
+        if (op3 == 0x0f)
+            val |= 0x01;
+    }
+    *field_ptr = val;
+    field_ptr++;
+    /* third evex byte */
+    val = (TEST(PREFIX_REX_W, di->prefixes) ? 0x80 : 0x00);
+    /* we override OPCODE_MODRM for vex to mean "requires vex.W" */
```

Comment:
You mean evex?  Seems like a bunch of copy-base from encode_vex_prefixes.

Reply:
Yes and yes. Fixed.